### PR TITLE
Add value sourcecontroller.extraEnv

### LINKED ISSUE #63: Configure proxy for source controller
### Is your feature request related to a problem ?

We need configure `https_proxy` for source controller to get those sources, but the deployment does't accept extra env vars.

### Describe the solution you'd like.

Add value `extraEnv` in the controller deployment.

### Describe alternatives you've considered.

None

### Additional context.

_No response_

### PR DESCRIPTION
**Pull request by uqix**
#### Which issue this PR fixes
  - fixes #63

#### Checklist

- [x] [DCO](https://github.com/fluxcd-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] helm-docs are updated
- [x] Helm chart is tested
- [x] Run `make reviewable`

### CODE REVIEW

**Comment by pierluigilenoci** [comment]
@uqix the DCO is missing

**Comment by uqix** [comment]
@pierluigilenoci DCO added

**Comment by dwerder** [comment]
@uqix can you run `make reviewable` , so that the versions in the test files and Chart.yaml are set correctly, please?

File: charts/flux2/templates/source-controller.yaml [hunk 1]
```diff
@@ -48,6 +48,9 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: metadata.namespace
+        {{- with .Values.sourcecontroller.extraEnv }}
+          {{- toYaml . | nindent 8 }}
```
Comment (dwerder) [inline charts/flux2/templates/source-controller.yaml:52]: nitpick, but can you remove two spaces to align it? Like in line 103. :)

**Comment by uqix** [comment]
@dwerder `make reviewable` done

**Comment by dwerder** [comment]
If test run fine. Can you please squash the commits to one?
```
git reset --soft HEAD~3
git commit -sm "Add value sourcecontroller.extraEnv"
git push
```

**Comment by dwerder** [comment]
LGTM

**Pull request merged**